### PR TITLE
Add the ability to specify the initial capacity of the ArrayNode

### PR DESCRIPTION
**Pull request by mveitas**
Addresses https://github.com/FasterXML/jackson-databind/issues/1181

### CODE REVIEW

**Comment by mveitas** [comment]
@cowtowncoder Would love to get your thoughts on this

File: src/main/java/com/fasterxml/jackson/databind/node/ArrayNode.java [hunk 1]
```diff
@@ -30,6 +30,11 @@ public ArrayNode(JsonNodeFactory nf) {
         _children = new ArrayList<JsonNode>();
```
Comment (mveitas) [inline src/main/java/com/fasterxml/jackson/databind/node/ArrayNode.java]: I thought about chaining the old constructor call to call the new constructor and provide a reasonable default of size 10 (what the default implementation is for an `ArrayList`).

Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/node/ArrayNode.java]: Probably fine to leave as is, not much benefit as far as I can see.

**Comment by cowtowncoder** [comment]
Looks simple enough to me. As to ctor chaining; I usually like chaining, but I think here it makes sense to just leave existing one as is, not much to win there, just adds one hop (which JVM may well inline, but possibly not).

So, I'd be happy to merge this in. The one formality I just need before that is the CLA (from https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf), unless I have received one earlier (tried to check but didn't see one -- unless you would be covered by a CCLA?).
The usual way is to print, fill + sign, scan and email it to `info` at fasterxml dot com.
If you could do that, I can go right ahead and merge this in. This is also the first and last time CLA needs to be provided, it's good for all the Jackson contributions.

Thanks in advance.

**Comment by mveitas** [comment]
Do you have a good process for porting a change like to back to other versions? We are currently using 2.5.4 and it would be nice to be able to make this available for 2.5.x

**Comment by cowtowncoder** [comment]
@mveitas Since this is API change, unfortunately it can not be backported to older versions.
In addition it is not clear whether new versions of 2.5 (or 2.6) will be released, except for possible hot-patches for specific component (like 2.5.5.1)

**Pull request merged**